icu4c/source/tools/toolutil/toolutil.cpp

```diff
@@ -218,6 +219,37 @@ uprv_fileExists(const char *file) {
 }
 #endif
 
+U_CAPI int32_t U_EXPORT2
+uprv_compareGoldenFiles(
+        const char* buffer, int32_t bufferLen,
+        const char* goldenFilePath,
+        bool overwrite) {
+
+    if (overwrite) {
+        std::ofstream ofs;
+        ofs.open(goldenFilePath);
+        ofs.write(buffer, bufferLen);
+        ofs.close();
+    }
+
+    std::ifstream ifs(goldenFilePath, std::ifstream::in);
+    char c = ifs.get();
+    int32_t pos = 0;
+    while (ifs.good() && pos < bufferLen) {
```

Comment:
I suggest checking for eof() rather than good(). At least, use eof() for end-of-file. If you like, you could also use good() or fail() for other errors.

Reply:
Done